modules/test/playwright/tests/object-web/rootModels.spec.ts

```diff
@@ -541,3 +541,214 @@ test.describe('Manage root models elements through Objects Admin', () => {
+
+		await modelBuilderRightSidebarPage.inheritanceCheckbox.click();
+
+		await expect(
+			objectRelationshipsPage.inheritanceModalHeader
+		).toBeVisible();
+
+		await objectRelationshipsPage.inheritanceModalDisableButton.click();
+
+		await modelBuilderRightSidebarPage.inheritanceCheckbox.waitFor();
+
+		await expect(
+			modelBuilderRightSidebarPage.inheritanceCheckbox
+		).not.toBeChecked();
+
+		await expect(
+			page
+				.getByRole('button', {name: objectRelationship.label['en_US']})
+				.getByRole('presentation')
+		).not.toBeVisible();
+
+		await expect(
+			modelBuilderDiagramPage.objectDefinitionNodes
+				.filter({
+					hasText: objectDefinition2.label['en_US'],
+				})
+				.locator('.lfr-objects__model-builder-node-container')
+		).not.toHaveClass(/treeItem/);
+	});
+
+	test('can create relationship with inheritance enabled using the add relationship modal', async ({
+		addNewObjectRelationshipModalPage,
+		apiHelpers,
+		modelBuilderDiagramPage,
+		modelBuilderRightSidebarPage,
+		page,
+	}) => {
+		const objectFolder =
+			await apiHelpers.objectAdmin.postRandomObjectFolder();
+
+		apiHelpers.data.push({id: objectFolder.id, type: 'objectFolder'});
+
+		const objectRelationships: ObjectRelationship[] = [];
+
+		try {
+			const objectDefinition1 =
+				await apiHelpers.objectAdmin.postRandomObjectDefinition(
+					{code: 0},
+					undefined,
+					objectFolder.externalReferenceCode
+				);
+
+			const objectDefinition2 =
+				await apiHelpers.objectAdmin.postRandomObjectDefinition(
+					{code: 0},
+					undefined,
+					objectFolder.externalReferenceCode
+				);
+
+			apiHelpers.data.push({
+				id: objectDefinition1.id,
+				type: 'objectDefinition',
+			});
+			apiHelpers.data.push({
+				id: objectDefinition2.id,
+				type: 'objectDefinition',
+			});
+
+			await modelBuilderDiagramPage.goto({
+				objectFolderName: objectFolder.name,
+			});
+
+			await modelBuilderDiagramPage.connectObjectDefinitionsNodeHandles(
+				objectDefinition1.id,
+				objectDefinition2.id
+			);
+
+			const objectRelationship =
+				await addNewObjectRelationshipModalPage.handleForm({
+					inherited: true,
+					objectRelationshipLabel:
+						'objectRelationship' + getRandomInt(),
+					type: 'One to Many',
+				});
+
+			objectRelationships.push(objectRelationship);
+
+			apiHelpers.data.push({
+				id: objectRelationship.id,
+				type: 'objectRelationship',
+			});
+
+			await expect(
+				page.locator('#ToastAlertContainer .alert-success')
+			).toBeVisible();
+
+			await expect(
+				modelBuilderRightSidebarPage.inheritanceCheckbox
+			).toBeChecked();
+		}
+		finally {
+			const objectRelationshipApiClient =
+				await apiHelpers.buildRestClient(ObjectRelationshipApi);
+
+			for (const objectRelationship of objectRelationships) {
+				await objectRelationshipApiClient.putObjectRelationship(
+					objectRelationship.id,
+					{
+						...objectRelationship,
+						edge: false,
```

Comment:
i didn't understand why this was necessary on this test, could you please explain? thanks!

Reply:
hey @rebsilva! it is needed for the clean up process to happen smoothly, if we don't unbind the relationship(set edge to `false`), it won't be deleted and throw an api error!

Reply:
i see, thanks for explaining! :blush: